docs/source/developers/docker.rst

```diff
@@ -0,0 +1,197 @@
+.. raw:: html
+
+   <!--
+   Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   -->
+
+Running Docker Builds
+=====================
+
+Most of our Linux based continuous integration tasks are decoupled from public
+CI services using docker and docker-compose. Keeping the CI configuration
+minimal makes local reproducibility possible.
+
+Usage
+-----
+
+There are multiple ways to execute the docker based builds. The recommended is
+to use the archery tool:
+
+Installation
+~~~~~~~~~~~~
+
+Requires ``python>=3.5``. It is recommended to install archery in ``editable``
+mode to automatically update the intallation by pulling the arrow repository.
+
+.. code:: bash
+
+    pip install -e dev/archery[docker]
+
+Inspect the available commands and options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: bash
+
+    archery docker --help
+    archery docker run --help
+
+``archery docker run`` tries to provide a similar interface to
```

Comment:
Does it though? `run` here seems to entail `build && run`.

Reply:
The command's interface tries to correspond to `docker-compose **run**` with the addition to build the required images. 
```bash

docker-compose run <image> <cmd> 
archery docker run <image> <cmd>

docker-compose run --env VAR_A=A -e VAR_B=B <image> <cmd> 
archery docker run --env VAR_A=A -e VAR_B=B <image> <cmd>
```

Reply:
Removed it.